src/components/colorBar/index.jsx

```diff
@@ -1,30 +1,31 @@
 import { useEffect, useRef } from "react";
-import Card from '@mui/material/Card';
 import Typography from '@mui/material/Typography';
 
 import { createColorbar } from "./helper";
 import * as d3 from "d3";
 
 import "./index.css";
 
-export const ColorBar = ({VMIN, VMAX, STEP, colorMap}) => {
+export const ColorBar = ({VMIN, VMAX, STEP, colorMap, skipStep=false, skipLabel=true}) => {
     const colorBarScale = useRef();
 
     useEffect(() => {
         const colorbar = d3.select(colorBarScale.current);
-        createColorbar(colorbar, VMIN, VMAX, STEP, colorMap);
+        createColorbar(colorbar, VMIN, VMAX, STEP, colorMap, skipStep);
```

Comment:
should this be inside `useMemo`?

Reply:
I don't think so. The first param to useMemo should be pure function, and here, we need to remove/cleanup the d3 dom element when the component unmounts. So they don't add up and hence is not compatible with useMemo.

Reply:
ah my bad, for some reason, i thought you were calculating the values of the colormap here